### PR TITLE
Feature: ability sign production targets during TUF root updates

### PR DESCRIPTION
**Pull request by vkhoroz**
Signed-off-by: Volodymyr Khoroz <volodymyr.khoroz@foundries.io>

### CODE REVIEW

**Comment by vkhoroz** [comment]
This works well with the most recent API changes, deployed to staging.

File: subcommands/keys/tuf_updates_sign_prod_targets.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		Long: `Sign production targets for your Factory with the offline targets key.
+New signatures are staged for commit together with TUF root modifications.
+
+There are 3 use cases when this command comes handy:
+- When you added a new offline TUF targets key,
+  and want to sign your Factory production targets with it.
+- When you increase the TUF targets signature threshold,
+  and need to sign your production targets with an additional key.
+- When you remove one of existing offline TUF targets keys,
+  and need to replace its signatures of production targets with signatures by another key.`,
```
Comment (doanac) [inline subcommands/keys/tuf_updates_sign_prod_targets.go:30]: @kprosise - can you give us a little grammar help?

File: subcommands/keys/tuf_utils.go [hunk 2]
```diff
@@ -410,6 +412,9 @@ func signProdTargets(factory string, signer TufSigner) (map[string][]tuf.Signatu
 
 	signatureMap := make(map[string][]tuf.Signature)
 	for tag, targets := range targetsMap {
+		if excludeFunc(tag, targets) {
```
[the remark this answers is not in the document]
Comment (doanac) [reply subcommands/keys/tuf_utils.go:415]: i guess you "fix" this in the next commit, but it still seems like we might want to be more defensive here or somethingl

File: subcommands/keys/tuf_updates_sign_prod_targets.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+func init() {
+	signCmd := &cobra.Command{
+		Use:   "sign-prod-targets --txid=<txid> --keys=<tuf-root-keys.tgz>",
+		Short: "Sign production targets for your Factory with the offline targets key",
+		Long: `Sign production targets for your Factory with the offline targets key.
+New signatures are staged for commit together with TUF root modifications.
+
+There are 3 use cases when this command comes handy:
+- When you added a new offline TUF targets key,
+  and want to sign your Factory production targets with it.
```
Comment (kprosise) [inline subcommands/keys/tuf_updates_sign_prod_targets.go:26]:
```suggestion
- You want to sign your Factory's production targets with a newly added offline TUF Targets key.
```

File: subcommands/keys/tuf_updates_sign_prod_targets.go [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+- When you increase the TUF targets signature threshold,
```
Comment (kprosise) [inline subcommands/keys/tuf_updates_sign_prod_targets.go:27]:
```suggestion
- You increase the TUF Targets signature threshold
```

File: subcommands/keys/tuf_updates_sign_prod_targets.go (hunk 1 above)
Comment (kprosise) [reply subcommands/keys/tuf_updates_sign_prod_targets.go:30]: Okay!

**Review by kprosise — COMMENTED** [review]
Left some suggestions!

File: subcommands/keys/tuf_updates_sign_prod_targets.go (hunk 4 above)
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_sign_prod_targets.go:27]: hmm... I've never seen us writing `production targets` as `production Targets`.

I'd have to change quite a few existing Fioctl commands to accommodate this suggestion.
So, I prefer to keep it as is for now.

File: subcommands/keys/tuf_utils.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (vkhoroz) [reply subcommands/keys/tuf_utils.go]: fixed in force-push

File: subcommands/keys/tuf_updates_sign_prod_targets.go (hunk 4 above)
Comment (kprosise) [reply subcommands/keys/tuf_updates_sign_prod_targets.go]: That's fine and a good plan! I already have a task to cleanup the commands, so I will just do it then and keep it consistent!

Comment (vkhoroz) [reply subcommands/keys/tuf_updates_sign_prod_targets.go]: So, I left `targets` as `targets` and applied the remainder of your corrections.

File: subcommands/keys/tuf_updates_sign_prod_targets.go [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+  and need to sign your production targets with an additional key.
```
[the remark this answers is not in the document]
Comment (vkhoroz) [reply subcommands/keys/tuf_updates_sign_prod_targets.go:27]: nixed per above 👍

**Pull request merged**